src/lang/modifyAst/boolean.ts

```diff
@@ -167,6 +167,26 @@ export function findAllChildrenAndOrderByPlaceInCode(
   artifact: Artifact,
   artifactGraph: ArtifactGraph
 ): Artifact[] {
+  const pathToCompositeSolidMap: { [key: string]: string[] } = {}
+  for (const [id, artifact] of artifactGraph) {
+    if (artifact.type === 'compositeSolid') {
+      for (const pathId of artifact.solidIds) {
+        if (pathToCompositeSolidMap[pathId]) {
+          pathToCompositeSolidMap[pathId].push(id)
```

Comment:
Is it possible for a path to get used in multiple CSG ops? If so, `composite_solid_id: Option<ArtifactId>,` would need to get converted to `composite_solid_ids: Vec<ArtifactId>,`.

Reply:
yes, but not yet I don't think, it would be the `keepTool` situation.